components/admin/LoginForm.tsx

```diff
@@ -12,9 +13,11 @@ export const LoginForm = () => {
     formState: { errors },
   } = useForm<LoginSchemaType>();
   const [error, setError] = useState<string | null>(null);
+  const [loading, setLoading] = useState<boolean>(false);
   const router = useRouter();
 
   const onSubmit: SubmitHandler<LoginSchemaType> = (data) => {
+    setLoading(true);
     Axios.post("/api/login", data)
```

Comment:
and also a small suggestion, but not a must-have - when we refactor all the queries to React Query, we can consider doing so with the mutations too